src/openms/source/CHEMISTRY/ResidueDB.cpp

```diff
@@ -535,7 +535,8 @@ namespace OpenMS
         try
         {
           // terminal modifications don't apply to residues (side chain), so only consider internal ones
-          mod = ModificationsDB::getInstance()->getModification(modification, residue->getOneLetterCode(), ResidueModification::ANYWHERE);
+          static const ModificationsDB* mdb = ModificationsDB::getInstance();
+          mod = mdb->getModification(modification, residue->getOneLetterCode(), ResidueModification::ANYWHERE);
```

Comment:
honestly, I dont understand why that is the case, but it really saves some time. I did not see this effect for `ResidueDB` and other singletons, its strange. I think keeping a single static ptr is safe, as the database can be modified but the ptr location is constant. Still, I cant understand why this makes a difference - but we should also look in other locations for this!

Reply:
is this also in -O3 ? I toyed around a bit on godbolt and the default parameters seem to generate quite some different code (also depending on the optimization level)

Reply:
Yes, this is clearly the case in -O3. 

I looked a bit further into this and it seems to only happen to `ModificationDB`, see https://github.com/OpenMS/OpenMS/pull/4469 and the only difference that I found was the default parameters which does make a big difference in the assembly (see https://github.com/OpenMS/OpenMS/pull/4469#issuecomment-589467386), basically the compiler cannot optimize away the function call and needs to set up all default string arguments which is very costly.